float8_experimental/float8_tensor.py

```diff
@@ -112,43 +112,12 @@ def to_fp8_no_autograd(
     return Float8Tensor(bits_fp8, x_scale, x.dtype, mm_config=mm_config)
 
 
-def from_fp8_no_autograd(x: torch.Tensor) -> torch.Tensor:
-    """Convert a tensor from float8 without autograd
-
-    This function will handle 3 cases:
-        1. If the tensor is a DTensor, it will convert the inner tensor to the original precision
-        2. If the tensor is a Float8Tensor, it will convert the tensor to the original precision
-        3. If the tensor is a regular tensor, it will pass through this tensor
-
-    Args:
-        x: the tensor to convert
-    """
-
-    def to_original_precision(grad):
-        if isinstance(grad, Float8Tensor):
-            return grad.to_original_precision()
-        else:
-            return grad
-
-    if isinstance(x, DTensor):
```

Comment:
I recalled the main reason we had that special handling is that torch.compile (specifically `FakeTensor` rule) can't take nested subclass rule automatically. If we remove this wondering where we would put such a logic? 

cc @drisspg @bdhirsh

Reply:
if I understand the old state correctly:
1. `grad` is never a `Float8Tensor` in practice, because we always have the matmul output high precision
2. things already work without nested subclasses, which is why removing this function is fine

But, would be great to clarify ^

Reply:
Oh I checked the current workflow, we call:
1. cast_to_float8_e4m3fn
2. cast_to_float8_e5m2_bw
and it seems none of these calls would call into this from_fp8_no_autograd

so this should be fine

Reply:
actually it looks like I'm wrong, the `cast_to_float8_e4m3fn`'s backward would call this, but maybe it's a no-op even as of current state, as the output of the torch.scaled_mm in backward is fp32?